File: backend/app/schemas/suggestion.py
"""Pydantic schemas for resource suggestions (response models)."""

from uuid import UUID

from pydantic import BaseModel, Field


class ResourceSuggestionResponse(BaseModel):
    """Response schema for a resource suggestion."""

    resource_id: UUID = Field(..., description="ID of the suggested resource")
    resource_name: str = Field(..., description="Name of the resource")
    qualification_summary: str = Field(
        ...,
        description=(
            "Comma-separated summary of matrix qualifications "
            "(e.g. 'Series Alpha/Electrical, Series Beta/Mechanical'). Empty string when "
            "no matrix entries exist."
        ),
    )
    department: str = Field(..., description="Department of the resource")
    availability_status: str = Field(
        ...,
        description="Availability status: available, partially_available, unavailable",
    )
    average_free_capacity: float = Field(
        ..., description="Average free capacity as percentage"
    )
    reason: str = Field(..., description="Explanation of suitability")

File: frontend/src/types/suggestion.ts
/**
 * TypeScript types for resource suggestions.
 * Based on the backend schema in app/schemas/suggestion.py.
 */

export type AvailabilityStatus = 'available' | 'partially_available' | 'unavailable'

export interface ResourceSuggestion {
  resource_id: string
  resource_name: string
  /**
   * Comma-separated summary of matrix qualifications
   * (e.g. "Series Alpha/Electrical, Series Beta/Mechanical"). Empty if the person
   * has no matrix entries.
   */
  qualification_summary: string
  department: string
  availability_status: AvailabilityStatus
  average_free_capacity: number
  reason: string
}

export interface SuggestionParams {
  start_date: string
  end_date: string
  allocation_percent: number
}
